### PR TITLE
Mypy fixes part 4

### PR DESCRIPTION
**Pull request by jenshnielsen**
Brings down the number of mypy errors to ~ 24

### CODE REVIEW

File: plottr/node/grid.py [hunk 1]
```diff
@@ -308,6 +308,8 @@ def __init__(self, node: Node = None):
         self.optGetters = {
             'grid': self.getGrid,
         }
+        if self.widget is None:
+            raise RuntimeError
```
Comment (jenshnielsen) [inline plottr/node/grid.py]: convert to assert

**Comment by jenshnielsen** [comment]
@astafan8 ready for review now. I have reverted the proplematic parts and will redo them in a new pr

File: plottr/data/datadict.py [hunk 2]
```diff
@@ -227,7 +227,7 @@ def add_meta(self, key: str, value: Any, data: Union[str, None] = None):
 
     set_meta = add_meta
 
-    def delete_meta(self, key, data=None):
+    def delete_meta(self, key: str, data=None) -> None:
```
Comment (astafan8) [inline plottr/data/datadict.py]:
```suggestion
    def delete_meta(self, key: str, data: Union[str, None] = None) -> None:
```

File: plottr/utils/num.py [hunk 3]
```diff
@@ -380,7 +384,8 @@ def crop2d(x: np.ndarray, y: np.ndarray, *arr: np.ndarray) \
     :return: all arrays (incl. x and y), cropped.
     """
     xs, ys = joint_crop2d_rows_cols(x, y)
-    allarrs = [] + [x, y] + [a for a in arr]
+    empty_list: List[np.ndarray] = []
+    allarrs: List[np.ndarray] = empty_list + [x, y] + [a for a in arr]
```
Comment (astafan8) [inline plottr/utils/num.py]: doesn't this work?
```suggestion
    allarrs = [x, y] + [a for a in arr]
```

File: plottr/plot/base.py [hunk 4]
```diff
@@ -41,7 +41,7 @@ def setPlotWidgetContainer(self, w: 'PlotWidgetContainer'):
         self.plotWidgetContainer = w
         self.newPlotData.connect(self.plotWidgetContainer.setData)
 
-    def process(self, dataIn: DataDictBase = None) -> Dict[str, DataDictBase]:
+    def process(self, dataIn: DataDictBase = None) -> Dict[str, Optional[DataDictBase]]:
```
Comment (astafan8) [inline plottr/plot/base.py]: ? or is it one of the problematic ones?
```suggestion
    def process(self, dataIn: Optional[DataDictBase] = None) -> Dict[str, Optional[DataDictBase]]:
```

**Review by astafan8 — DISMISSED** [review]

File: plottr/utils/num.py (hunk 3 above)
Comment (jenshnielsen) [reply plottr/utils/num.py]: I would say so but I always try to not make any code changes while adding types but this does look pointless

**Review by astafan8 — APPROVED** [review]

**Pull request merged**